tuplex/codegen/src/IteratorContextProxy.cc

```diff
@@ -0,0 +1,494 @@
+#include <IteratorContextProxy.h>
+
+namespace tuplex {
+    namespace codegen {
+
+        SerializableValue IteratorContextProxy::initIterContext(LambdaFunctionBuilder &lfb, llvm::IRBuilder<> &builder,
+                                                            const python::Type &iterableType,
+                                                            const SerializableValue &iterable) {
+            using namespace llvm;
+
+            if(iterableType == python::Type::EMPTYLIST || iterableType == python::Type::EMPTYTUPLE) {
```

Comment:
what about empty dict? If not supported, let's issue at least a warning.

Reply:
Dictionary is handled through fallback.
Added check and error message.

Reply:
hm, I would maybe add an assert check then here to make sure no type is dictionary type. This way, it's clear there's a gap in terms of implementation here. I.e., a function should never assume only "good" data is passed to it.

Reply:
Yes, a check has been added right after this line.